server/conn.go

```diff
@@ -37,50 +37,43 @@ type Conn struct {
 }
 
 var baseConnID uint32 = 10000
+var defaultServer *Server
 
 // NewConn: create connection with default server settings
+// Deprecated: Use Server.NewConn instead.
 func NewConn(conn net.Conn, user string, password string, h Handler) (*Conn, error) {
-	p := NewInMemoryProvider()
-	p.AddUser(user, password)
-
-	var packetConn *packet.Conn
-	if defaultServer.tlsConfig != nil {
-		packetConn = packet.NewTLSConn(conn)
-	} else {
-		packetConn = packet.NewConn(conn)
+	if defaultServer == nil {
+		defaultServer = NewDefaultServer()
 	}
+	return defaultServer.NewConn(conn, user, password, h)
+}
```

Comment:
Same remarks as above


```suggestion
// NewCustomizedConn: create connection with customized server settings
//
// Deprecated: Use [Server.NewConn] instead.
```